backend/src/main/java/com/github/sirtobiwan/backend/security/VaccineUserController.java

```diff
@@ -0,0 +1,41 @@
+package com.github.sirtobiwan.backend.security;
+
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("api/users")
+public class VaccineUserController {
+    private final VaccineUserDetailsService vaccineUserDetailsService;
+
+    public VaccineUserController(VaccineUserDetailsService vaccineUserDetailsService) {
+        this.vaccineUserDetailsService = vaccineUserDetailsService;
+    }
+
+    @GetMapping("me")
+    public String getUserInfo() {
+        return SecurityContextHolder
+                .getContext()
+                .getAuthentication()
+                .getName();
+    }
+
+    @PostMapping("/login")
+    public String login(){
+        return SecurityContextHolder
+                .getContext()
+                .getAuthentication()
+                .getName();
+    }
+
+    private VaccineUser convertDtoToEntity(DtoVaccineUser dto) {
+        return new VaccineUser(dto.id(), dto.username(), dto.password());
```

Comment:
Isn't `dto.id() == null` since you're not passing an `id` in the RequestBody?
That way the new VaccineUser is created with `id == null` before creating another new VaccineUser in the Service class. You could just pass the dto without an id to the service and create a new VaccineUser there.

Reply:
absolutely. I removed it. Thanks